argonaut/src/main/scala/org/http4s/argonaut/ArgonautInstances.scala

```diff
@@ -13,6 +13,9 @@ trait ArgonautInstances {
     jawn.jawnDecoder
 
   def jsonOf[F[_]: Sync, A](implicit decoder: DecodeJson[A]): EntityDecoder[F, A] =
+    jsonOf(decoder)
+
+  def jsonOf[F[_]: Sync, A](decoder: DecodeJson[A]): EntityDecoder[F, A] =
```

Comment:
Wow, so these overloads don't clash because of the desugaring of the `Sync` bound?  I had to stare at this for a couple minutes to figure out how it could compile. 😖 

Is this necessary to support an explicit `DecodeJson` as opposed to calling `jsonOf[Foo]`?

Reply:
Correct. If it helps I can make the implicit parameter, erm, explicit. :)